### PR TITLE
Add the giveaway UI

### PR DESCRIPTION
**Pull request by johnhforrest**
Users that login and have no records in their collection can claim a record from the giveaway.

### CODE REVIEW

File: src/views/RecordListView.vue [hunk 1]
```diff
@@ -1,14 +1,55 @@
...
     </app-header>
     <b-card-group deck class="record-list" v-if="records.length">
-      <record-list-item v-for="record in records"
+      <record-list-item
+        v-for="record in records"
         :codex-record="record"
         :key="record.tokenId"
       />
     </b-card-group>
+    <b-card-group deck class="record-list giveaway" v-else-if="giveaway">
```
Comment (Anaphase) [inline src/views/RecordListView.vue]: Should this `v-else-if` be a `v-if` instead? Dunno if the intent was to always show available giveaways, or only if you have no other Records.

File: src/views/RecordListView.vue [hunk 2]
```diff
@@ -74,12 +130,88 @@ export default {
...
+        })
+    },
+    createGiveaway() {
+      axios.post('/giveaways')
+        .then((response) => {
+          console.log(response)
+        })
+    },
+    acceptGiveaway() {
+      // TODO: Show spinner
```
Comment (Anaphase) [inline src/views/RecordListView.vue]: This TODO is done?

File: src/views/RecordListView.vue [hunk 3]
```diff
@@ -74,12 +130,88 @@ export default {
...
+      this.disableGiveawayButton = true
+      this.isLoading = true
+
+      axios.get(`/user/giveaway/${this.giveaway._id}`)
+        .then((response) => {
+          // TODO: Show some information dialog and stop the spinner
+          // this.isLoading = false
```
Comment (Anaphase) [inline src/views/RecordListView.vue]: I think this should be uncommented now that there's a spinner?

File: src/views/RecordListView.vue [hunk 4]
```diff
@@ -74,12 +130,88 @@ export default {
...
+  width: 100%
+  height: 100%
+  z-index: 998
+  display: flex
+  position: absolute
+  align-items: center
+  justify-content: center
+  background-color: rgba(255, 255, 255, .75)
+
+@keyframes spin {
```
Comment (Anaphase) [inline src/views/RecordListView.vue]: All these brackets & semicolons, oy vey!

```stylus
.spinner
  width: 8em
  height: 8em
  z-index: 999
  animation: spin 1s linear infinite reverse

@keyframes spin
  from
    transform: rotate(0deg)

  to
    transform: rotate(360deg)
```

File: src/views/RecordListView.vue (hunk 1 above)
Comment (johnhforrest) [reply src/views/RecordListView.vue]: For now only if you have no other records, which will be all new users in the beta.

File: src/views/RecordListView.vue (hunk 4 above)
Comment (johnhforrest) [reply src/views/RecordListView.vue]: hahaha. copy/pasted from biddable!

**Pull request merged**